### DISCUSSION TITLE
Passing params other than those in NavigatorScreenParams to a nested navigator

### DISCUSSION THREAD

**Discussion by tibbe** [react-navigation/react-navigation#11856] [Q&A]
According to https://reactnavigation.org/docs/typescript/#nesting-navigators we can type nested navigators using `NavigatorScreenParams`.

```tsx
type TabParamList= {
  // All screens have a common param, `x`
  One: {x: string, y: number};
  Two {x: string, z: boolean};
}

type StackParamList = {
  Home: NavigatorScreenParams<TabParamList>;
  Other: undefined;
}

export type StackScreenProps<T extends keyof StackParamList> =
  StackScreenProps<StackParamList, T>;

export type TabScreenProps<T extends keyof TabParamList> =
  CompositeScreenProps<
    NativeStackScreenProps<TabParamList, T>,
    StackScreenProps<keyof StackParamList>
  >;

function OneScreen({route}: TabScreenProps<'One'>) {
  return <Text>{route.x} - {route.y}</Text>
}


function TwoScreen({route}: TabScreenProps<'Two'>) {
  return <Text>{route.x} - {route.x}</Text>
}

const Tab = createBottomTabNavigator();

function TabNavigator({route}: StackScreenProps<'Home'>) {
  // Maybe do something with `x`, which should be present on all tabs.
  return (
    <Tab.Navigator>
      <Tab.Screen name="One" component={OneScreen} />
      <Tab.Screen name="Two" component={TwoScreen} />
    </Tab.Navigator>
  );
}
```

We can then navigate to one of the tabs:

```tsx
navigation.navigate('Home', {
  screen: 'One',
  params: { x: 'foo', y: 1.0 },
});
```

The issue is that if we tap the second tab ("Two") then we end up on that tab without its params (i.e. `x` and `z`) set. This makes sense for `z`, as there's no way to pass it in the tap, but I'd like for `x` to be set on every tab.

In other words, I want the tabs to have a set of common params (e.g. `x` above) that are always present no matter what tab you tap once you arrive on `MyTabNavigator` (through a `navigate()` call from some other screen).

How do I achieve this? What's the right way to pass params that apply to all "leaves" (and could potentially also be used in the implementation of the nested navigator itself e.g. to set its header)?

Can the type of `Home` above be something like `NavigatorScreenParams<HomeTabsParamList> & {x: string}` (assuming `x` is then removed from each tab screen)? Is a context (containing `x`) the solution?
